diff --git a/stdlib/Makefile b/stdlib/Makefile
index 5f7bde13e..590badff8 100644
--- a/stdlib/Makefile
+++ b/stdlib/Makefile
@@ -127,8 +127,8 @@ CAMLHEADERS =\
 ifeq "$(HASHBANGSCRIPTS)" "true"
 $(CAMLHEADERS): ../config/Makefile
 	for suff in '' d i; do \
-	  echo '#!$(BINDIR)/ocamlrun'$$suff > camlheader$$suff && \
-	  echo '#!$(TARGET_BINDIR)/ocamlrun'$$suff >target_camlheader$$suff; \
+	  echo '#!/usr/bin/env ocamlrun'$$suff > camlheader$$suff && \
+	  echo '#!/usr/bin/env ocamlrun'$$suff >target_camlheader$$suff; \
 	done && \
 	echo '#!' | tr -d '\012' > camlheader_ur;
 else # Hashbang scripts not supported
