--- a/version.rc
+++ b/version.rc
@@ -21,8 +21,8 @@
     BEGIN
         BLOCK "040904b0"
         BEGIN
-            VALUE "FileVersion", FLEXDLL_FULL_VERSION
-            VALUE "ProductVersion", FLEXDLL_FULL_VERSION
+            VALUE "FileVersion", "0.44"
+            VALUE "ProductVersion", "0.44"
             VALUE "ProductName", "FlexDLL"
             VALUE "FileDescription", "FlexDLL Linker"
             VALUE "LegalCopyright", "Institut National de Recherche en Informatique et en Automatique"
