--- a/Makefile
+++ b/Makefile
@@ -158,7 +158,7 @@
 build_msvc64: flexdll_msvc64.obj flexdll_initer_msvc64.obj
 build_cygwin64: flexdll_cygwin64.o flexdll_initer_cygwin64.o
 build_mingw: flexdll_mingw.o flexdll_initer_mingw.o
-build_mingw64: flexdll_mingw64.o flexdll_initer_mingw64.o
+build_mingw64: flexdll_mingw64.o flexdll_initer_mingw64.o flexdll_mingw64.obj flexdll_initer_mingw64.obj ../flexdll_mingw64.o ../flexdll_initer_mingw64.o
 build_mingw64arm: flexdll_mingw64arm.obj flexdll_initer_mingw64arm.obj

 OBJS = version.ml Compat.ml coff.ml cmdline.ml create_dll.ml reloc.ml
@@ -263,6 +263,18 @@
 flexdll_initer_mingw64.o: flexdll_initer.c
 	$(MIN64CC) $(GCC_FLAGS) -c -o $@ $<

+flexdll_mingw64.obj: flexdll_mingw64.o
+	cp $< $@
+
+flexdll_initer_mingw64.obj: flexdll_initer_mingw64.o
+	cp $< $@
+
+../flexdll_mingw64.o: flexdll_mingw64.o
+	cp $< $@
+
+../flexdll_initer_mingw64.o: flexdll_initer_mingw64.o
+	cp $< $@
+
 flexdll_initer_mingw64arm.obj: flexdll_initer.c
 	$(CC) $(GCC_FLAGS) -c -o $@ $<

